server/rest/src/main/java/org/hawkular/apm/server/rest/AnalyticsHandler.java

```diff
@@ -89,1578 +77,289 @@
+                    request.getStartTime(),
+                    request.getEndTime(),
+                    request.isCompress()
+            );
+            log.tracef("Got unbound endpoints: start [%s] end [%s] = [%s]", request.getStartTime(), request.getEndTime(), endpoints);
+
+            return Response
+                    .status(Response.Status.OK)
+                    .entity(endpoints)
+                    .type(APPLICATION_JSON_TYPE)
+                    .build();
+        });
     }
 
     @GET
     @Path("boundendpoints/{name}")
-    @Produces(APPLICATION_JSON)
-    @ApiOperation(
-            value = "Identify the bound endpoints for a business transaction",
-            response = List.class)
+    @ApiOperation(value = "Identify the bound endpoints for a business transaction", response = List.class)
     @ApiResponses(value = {
             @ApiResponse(code = 200, message = "Success"),
             @ApiResponse(code = 500, message = "Internal server error") })
-    public void getBoundEndpoints(
-            @Context SecurityContext context, @HeaderParam("Hawkular-Tenant") String tenantId,
-            @Suspended final AsyncResponse response,
-            @ApiParam(required = true,
-            value = "business transaction name") @PathParam("name") String name,
-            @ApiParam(required = false,
-            value = "optional 'start' time, default 1 hour before current time") @DefaultValue("0")
-            @QueryParam("startTime") long startTime,
-            @ApiParam(required = false,
-            value = "optional 'end' time, default current time") @DefaultValue("0") @QueryParam("endTime")
-            long endTime) {
-
-        try {
-            log.tracef("Get bound endpoints: name [%s] start [%s] end [%s]", name, startTime, endTime);
-
-            java.util.List<EndpointInfo> endpoints = analyticsService.getBoundEndpoints(
-                    securityProvider.validate(tenantId, context.getUserPrincipal().getName()), name, startTime, endTime);
-
-            log.tracef("Got bound endpoints: name [%s] start [%s] end [%s] = [%s]", name, startTime, endTime,
-                                                endpoints);
-
-            response.resume(Response.status(Response.Status.OK).entity(endpoints).type(APPLICATION_JSON_TYPE)
-                    .build());
-
-        } catch (Throwable e) {
-            log.debug(e.getMessage(), e);
-            Map<String, String> errors = new HashMap<String, String>();
-            errors.put("errorMsg", "Internal Error: " + e.getMessage());
-            response.resume(Response.status(Response.Status.INTERNAL_SERVER_ERROR)
-                    .entity(errors).type(APPLICATION_JSON_TYPE).build());
-        }
-
+    public Response getBoundEndpoints(@BeanParam BoundEndpointsRequest request) {
+        return returnOrCatch(() -> {
+            log.tracef("Get bound endpoints: name [%s] start [%s] end [%s]", request.getName(), request.getStartTime(), request.getEndTime());
```

Comment:
For the future, we might want to remove those logs, making the services as EJBs and using interceptors to do such calls. This cleans up the business code, and provides a standard logging for all business methods.